Sources/DBSCAN/DBSCAN.swift

```diff
@@ -82,14 +82,16 @@ public struct DBSCAN<Value: Equatable> {
 
         var clusters: [[Value]] = []
         var outliers: [Value] = []
-
-        for points in Dictionary(grouping: points, by: { $0.label }).values {
```

Comment:
I think we can avoid having to call `filter` against a list of points by using the constructed dictionary to fuller effect. Would something like this work instead?

```swift
for (label, points) in Dictionary(grouping: points, by: { $0.label }) {
    if label == nil {
        outliers.append(contentsOf: values)
    } else {
        clusters.append(values)
    }
}
```

Reply:
Thanks for great comment!
I changed my code and make sure it works.